src/core/annotation.js

```diff
@@ -752,6 +754,26 @@ var TextWidgetAnnotation = (function TextWidgetAnnotationClosure() {
   return TextWidgetAnnotation;
 })();
 
+var ButtonWidgetAnnotation = (function ButtonWidgetAnnotationClosure() {
+  function ButtonWidgetAnnotation(params) {
+    WidgetAnnotation.call(this, params);
+
+    this.data.noToggleToOff = this.hasFieldFlag(
+      AnnotationFieldFlag.NOTOGGLETOOFF);
+    this.data.radio = this.hasFieldFlag(AnnotationFieldFlag.RADIO);
```

Comment:
Well, PDF.js is a *PDF viewer* after all, so we most certainly want to try and follow the specification!
If experience has thought us anything, it's that PDF files found in the wild breaks the specification in all sorts of "interesting" ways, which often leads to bugs. So of course we always want try to ensure that data present in a PDF file is valid!